cmake/strip.cmake

```diff
@@ -0,0 +1,18 @@
+# * Strips debug symbols from an executable target
+# * Copies debug symbols into an extra debug-symbol-file
+# * Adds reference to this debug symbol from to the executable
+# * Does not support Windows / Is not needed on Windows
+function(strip_symbols target)
+  if(NOT WIN32)
+    add_custom_command(
+            TARGET ${target} POST_BUILD
+      COMMAND ${CMAKE_OBJCOPY} --only-keep-debug "$<TARGET_FILE_NAME:${target}>"
+              "$<TARGET_FILE_NAME:${target}>.debug"
+      COMMAND ${CMAKE_OBJCOPY} --strip-debug "$<TARGET_FILE_NAME:${target}>"
+      COMMAND
+        ${CMAKE_OBJCOPY}
+        --add-gnu-debuglink="$<TARGET_FILE_NAME:${target}>.debug"
+        "$<TARGET_FILE_NAME:${target}>"
```

Comment:
I guess this means I just need to push both files and debuglink should instruct gdb to do the right thing...

Reply:
Yes, exactly. As far as I understood it you can either provide an absolute or a relative path.
Linux distros usually use absolute paths and have there debug files in one place (i.e. /usr/lib/debug). I think for our use case relative paths (relative to the executable) is probably more appropriate.